src/app.jsx

```diff
@@ -12,6 +13,55 @@ import routes from './routes';
 import analytics from './services/analytics';
 
 const App = () => {
+  // Code for setting up color schemes
+
+  const { mode, setMode } = useColorScheme();
+  // mode listener code from
+  //https://dev.to/gustavogomez/manage-dark-mode-in-reactjs-with-listener-to-detect-dark-mode-in-the-browser-3120
+
+  let colorMode = localStorage.getItem('colorMode') ?? 'system';
+
+  const onSelectMode = (prefersDark) => {
+    prefersDark === 'dark' ? setMode('dark') : setMode('light');
+  };
+
+  const setupColorSchemeListeners = (colorScheme) => {
+    if (colorMode === 'system' || colorMode === null) {
+      //anything other than 'dark' will set the scheme to light
+      setMode(colorMode);
+
+      //listen for system setting changes
+      window
+        .matchMedia('(prefers-color-scheme: dark)')
+        .addEventListener('change', (e) => onSelectMode(e.matches ? 'dark' : 'light'));
+    } else {
+      //stop listening for system setting changes and set the mode
+      window.matchMedia('(prefers-color-scheme: dark)').removeEventListener('change', () => {});
+      setMode(colorMode);
+    }
+  };
+
+  //Callback, when the colorScheme changes in local storage
+  const onModeUpdate = () => {
+    setupColorSchemeListeners(localStorage.getItem('colorMode'));
+  };
+
+  useEffect(() => {
+    //Listen for changes in local storage
+    window.addEventListener('storage', onModeUpdate);
+    setupColorSchemeListeners(colorMode);
```

Comment:
You aren't fully utilizing events here - you wait for them to trigger, but you don't actually read the data they contain.

```suggestion
  useEffect(() => {
    //Callback, when the colorScheme changes in local storage
    const onModeUpdate = (storageEvent) => {
      if (storageEvent.key === 'colorMode') {
        setupColorSchemeListeners(storageEvent.newValue);
      }
    };

    //Listen for changes in local storage
    window.addEventListener('storage', onModeUpdate);
    setupColorSchemeListeners(localStorage.getItem('colorMode'));
```

Reply:
Fixed by the custom hook